Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityOptimizeWindow.cs

```diff
@@ -0,0 +1,607 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    /// <summary>
+    /// Build window - Utility for developers to automate optimization of their Unity scene/project
+    /// </summary>
+    public class MixedRealityOptimizeWindow : EditorWindow
+    {
+        private int selectedToolbarIndex = 0;
+        private readonly string[] ToolbarTitles = { "Project Optimizations", "Scene Optimizations", "Shader Optimizations" };
+        private enum ToolbarSection
+        {
+            Project = 0,
+            Scene,
+            Shader
+        };
+
+        private Vector2 windowScrollPosition = new Vector2();
+        private bool showProjectOptimizations = true;
+        private bool showSceneOptimizations = true;
+        private bool showShaderOptimizations = true;
+
+        // Project Optimizations
+        private bool singlePassInstanced = true;
+        private bool enableDepthBufferSharing = true;
+        private bool enable16BitDepthBuffer = true;
+
+        // Scene Optimizations
+        private bool disableRealtimeGlobalIllumination = true;
+        private bool disableBakedGlobalIllumination = true;
+        private static DateTime? lastAnalyzedTime = null;
+        private Light[] sceneLights;
+        private const uint TopListSize = 5;
+        private long totalActivePolyCount, totalInActivePolyCount = 0;
+        private long totalPolyCount
+        {
+            get
+            {
+                return totalActivePolyCount + totalInActivePolyCount;
+            }
+        }
+        private string TotalPolyCountStr, TotalActivePolyCountStr, TotalInactivePolyCountStr;
+        private GameObject[] LargestMeshes = new GameObject[TopListSize];
+        private int[] LargestMeshSizes = new int[TopListSize];
+
+        // Shader Optimizations
+        private bool showDiscoveredMaterials = true;
+        private bool onlyUnityShader = true;
+        private Shader replacementShader;
+        private Shader unityStandardShader;
+        private Shader errorShader;
+        private List<Material> discoveredMaterials = new List<Material>();
+
+        // Internal structure to easily search mesh polycounts in scene
+        private struct MeshNode
+        {
+            public int polycount;
+            public MeshFilter filter;
+        }
+
+        protected enum PerformanceTarget
```

Comment:
Super dee duper nit: Not loving the underscores in the enum values. They look ugly.